gprofiler/application_separators.py

```diff
@@ -0,0 +1,238 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_bin(bin_name: str):
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
+    @abstractmethod
+    def is_supported(self, process: Process) -> bool:
+        pass
+
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def app_prefix(self) -> str:
+        pass
+
```

Comment:
As we talked yesterday, in many cases one of these functions is redundant so my suggestion is to have a single `get_application_name` method that'll perform the formatting internally

Reply:
Done, resolving